### DISCUSSION TITLE
How to generate token by token and api

### DISCUSSION THREAD

**Discussion by johnwick123f** [abetlen/llama-cpp-python#404] [Q&A] [answered]
I used the demo code and it worked! but I have 2 questions. First, how do you generate the response by token so the response doesn’t come out all at once? And second, what does it mean by high level api? Is it just some code or does it use the model on some servers?

**Comment by AnOpenSauceDev** [comment, marked as answer]
> I used the demo code and it worked! but I have 2 questions. First, how do you generate the response by token so the response doesn’t come out all at once? And second, what does it mean by high level api? Is it just some code or does it use the model on some servers?

`stream=true` should do the trick, and high-level basically means it's an abstraction to make development more simple.
